### PR TITLE
Add digital ocean guide.

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

File: master/reference/public-cloud/digital-ocean.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Calico can be launched on either:
+
+- Public Interface
+- Private Interface [Optional]
+
+We recommend running Calico on the private interface, as it enhances security by
+ensuring Calico services are not accessible from the wider internet.
+
+However, hosts in different Datacenter Regions will not have IP connectivity with one
+another on their private address, and therefore will not be able to establish
```
Comment (gunjan5) [inline master/reference/public-cloud/digital-ocean.md]: -> "another through their private address, and therefore they will not be able to establish"

File: master/reference/public-cloud/digital-ocean.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+BGP sessions with one another. For mluti-region digital ocean clusters,
```
Comment (gunjan5) [inline master/reference/public-cloud/digital-ocean.md]: "digital ocean" -> "Digital Ocean"

File: master/reference/public-cloud/digital-ocean.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+bind Calico to the public interface.
+
+###### Enable Encapsulation
+
+In Digital Ocean, cross-host container-to-container traffic will travel over an
+L3 hop. Since Digital Ocean does not allow peering to the L3 hop, it will
```
Comment (gunjan5) [inline master/reference/public-cloud/digital-ocean.md]: "L3 hop" doesn't sound right. There could be more than 1 L3 hops involved in this.  Maybe call it "L3 network"?

File: master/reference/public-cloud/digital-ocean.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+not know how to route this container traffic, and will drop it.
+
+To remedy this, Calico can encapsulate container traffic with the IP so that the
+networking fabric never sees the container IPs, allowing standard routing to
+take over. Turn on traffic encapsulation in pool settings by enabling:
+
+- `ipip` for container-to-container traffic.
+- `nat-outgoing` for container-to-ec2-instance traffic.
```
Comment (gunjan5) [inline master/reference/public-cloud/digital-ocean.md]: ec2?

File: master/reference/public-cloud/digital-ocean.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+See [pool configuration]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/ippool) for information on how to enable this for IP pools.
+
+###### Ensure Calico Chooses Correct IP
+
+Note that calicoctl's automatic IP detection will launch calico/node on the private
+address if the droplet is configured to have a private interface.
+
+Ensure you manually specify `--ip` when launching `calicoctl node run` if you
+want it to bind to the public interface.
```
Comment (gunjan5) [inline master/reference/public-cloud/digital-ocean.md]: might be a good idea to link to `calicoctl node run` docs page here

File: master/reference/public-cloud/digital-ocean.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+Calico is designed to provide high performance massively scalable virtual
+networking for containers in public cloud environments such as Digital Ocean.
+
+## How to Run Calico in Digital Ocean
+
+Digital Ocean offers two networking options for droplets.
+Calico can be launched on either:
+
+- Public Interface
+- Private Interface [Optional]
```
Comment (gunjan5) [inline master/reference/public-cloud/digital-ocean.md]: not sure I understand why there's an `[Optional]` tag here. You can have a Calico network completely on the Private interface, same goes for Public interface, `[Optional]` suggests Public one is mandatory but Private is not. WDYT?

Comment (djosborne) [inline master/reference/public-cloud/digital-ocean.md]: In digital ocean, all droplets have a public interface, but you can optionally add a private interface. I'll clarify that that's what I'm referring to here.

**Comment by dghubble** [comment]
Why was this closed?

**Comment by ozdanborne** [comment]
@dghubble no particular reason. Calico works on DO, so there should be no problem. The doc just never got in for some reason.

If you're having trouble deploying there, do open a github issue

**Comment by dghubble** [comment]
Thanks, filed #1095

**Pull request closed without merging**